url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/io/image/TargaImage.java

```diff
@@ -0,0 +1,345 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import java.awt.Dimension;
+import java.awt.image.DirectColorModel;
+
+import vsdk.toolkit.io.PersistenceElement;
+import vsdk.toolkit.media.RGBImage;
+import vsdk.toolkit.media.RGBAImage;
+
+/**
+The TargaImage class represents a targa image (the image files with .tga
+extension)
+*/
+
+public class TargaImage extends PersistenceElement
+{
+    private static final int NO_TRANSPARENCY = 255;
+    private static final int FULL_TRANSPARENCY = 0;
+    
+    private short idLength;
+    private short colorMapType;
+    private short imageType;
+    private int cMapStart;
+    private int cMapLength;
+    private short cMapDepth;
+    private int xOffset;
+    private int yOffset;
+    private int xSize;
+    private int ySize;
+    private short pixelDepth;
+    private short imageDescriptor;
+    private DirectColorModel cm;
+    public byte[] pixels;
+
+    /**
+    Constructs a TGAImage given the File image
+    @param srcFile The tga image file
+    */
+    public TargaImage(File srcFile) throws ImageNotRecognizedException {
+        try
+        {
+            open(srcFile);
+        }
+        catch(IOException ioe)
+        {
+            throw new ImageNotRecognizedException("I/O Error while reading the image", srcFile);
+        }
+    }
+    
+    private void open(File srcFile) throws ImageNotRecognizedException, IOException {
+        byte red = 0;
+        byte green = 0;
+        byte blue = 0;
+        byte alpha = FULL_TRANSPARENCY;
+        int srcLine = 0;
+        
+        FileInputStream fis = new FileInputStream(srcFile);
+        BufferedInputStream bis = new BufferedInputStream(fis, 8192);
+        DataInputStream dis = new DataInputStream(bis);
+        
+        idLength = (short) dis.read();
+        colorMapType = (short) dis.read();
+        imageType = (short) dis.read();
+        cMapStart = (int) flipEndian(dis.readShort());
+        cMapLength = (int) flipEndian(dis.readShort());
+        cMapDepth = (short) dis.read();
+        xOffset = (int) flipEndian(dis.readShort());
+        yOffset = (int) flipEndian(dis.readShort());
+        xSize = (int) flipEndian(dis.readShort());
+        ySize = (int) flipEndian(dis.readShort());
+        pixelDepth = (short) dis.read();
+        
+        if (pixelDepth == 24) 
+        {
+            cm = new DirectColorModel(24, 0xFF0000, 0xFF00, 0xFF);
+            pixels = new byte[xSize * ySize*3];
+        } 
+        else { 
+            if ( pixelDepth == 32 ) {
+                cm = new DirectColorModel(32, 0xFF000000, 0xFF0000, 0xFF00, 0xFF);
+                pixels = new byte[xSize * ySize*4];
+            }
+            else {
+                System.err.println("TargaImage - cannot read file!");       
+            }
+        }
+        
+        imageDescriptor = (short) dis.read();
+        
+        if ( idLength > 0 ) {
+            bis.skip(idLength);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Check the number of bytes that were skipped when you called `skip()`. Handle cases when fewer than the expected number of bytes were skipped. See the following diffs on GitHub that make this code change: [diff1](https://github.com/apache/tomcat80/commit/facf983d8a7952e8d6b03204cb6610fc5f981dc6#diff-e58d7c084e48bd454de5beb71712f72d243bd0287ae2096d7b771f3a2dae6fb0R174), [diff2](https://github.com/apache/hadoop/commit/45620eee687b1093c812873a99833dab068e43d2#diff-f1ae2a4b3016a9deb983f2562d1e71372baf960cf5122d474a37a00f0307edb2R45)

Learn more: [InputStream.skip in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#skip-long-)